test/powershell/Modules/Microsoft.WSMan.Management/ConfigProvider.Tests.ps1

```diff
@@ -0,0 +1,441 @@
+Describe "WSMan Config Provider" -Tag Feature,RequireAdminOnWindows {
+    BeforeAll {
+        #skip all tests on non-windows platform
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        $PSDefaultParameterValues["it:skip"] = !$IsWindows
+
+        if ($IsWindows) {
+            $badCredentialError = 1326
+            $endpoint = & "$PSHOME\Install-PowerShellRemoting.ps1"
+            $pluginXml = [xml](winrm g winrm/config/plugin?name=$($endpoint.name) -format:xml)
+            $pluginPath = "WSMan:\localhost\Plugin\$($endpoint.name)"
+            $testPluginXml = [xml]($pluginXml.OuterXml)
+            $testPluginXml.PlugInConfiguration.Name = "TestPlugin"
+            $testPluginXml.PlugInConfiguration.RemoveAttribute("xml:lang")
+            $testPluginXml.PlugInConfiguration.Resources.Resource.ResourceUri = "http://schemas.microsoft.com/powershell/TestPlugin"
+            $testPluginXml.PlugInConfiguration.Resources.Resource.Security.Uri = "http://schemas.microsoft.com/powershell/TestPlugin"
+            Set-Content $TestDrive\plugin.xml -Value $testPluginXml.OuterXml
+            $null = winrm c winrm/config/plugin?Name=TestPlugin -file:$TestDrive\plugin.xml
+            $testPluginPath = "WSMan:\localhost\Plugin\TestPlugin"
+            $testUser = (Get-Random)
+            $testPass = "Secret123!"
+            $null = net user $testUser $testPass /add
+        }
+    }
+
+    AfterAll {
+        $PSDefaultParameterValues = $originalDefaultParameterValues
+        if ($IsWindows) {
+            $null = winrm d winrm/config/plugin?Name=TestPlugin
+            $null = net user $testUser /DELETE
+        }
+    }
+
+    Function Test-Plugin($plugin, $expectedMissingProperties, $expectedMissingAttributes) {
+        $plugin.PSPath | Should Exist
+        $plugin | Should Not BeNullOrEmpty
+        $testPluginXml = [xml](winrm g winrm/config/plugin?name=$($plugin.Name) -format:xml)
+        $pluginProperties = Get-ChildItem $plugin.PSPath
+        $pluginProperties.Count | Should Be (($testPluginXml.PluginConfiguration | Get-Member -Type Properties).Count + $expectedMissingProperties.Count - $expectedMissingAttributes.Count)
```

Comment:
Please move the value calculated for Should Be to a variable. It will make it more readable.

Reply:
ok